ui/src/model/mobileapp.js

```diff
@@ -0,0 +1,79 @@
+export class MobileApp {
+  constructor(json) {
+    this.app = json;
+    this.spec = new MobileAppSpec(this.app);
+    this.metadata = new MobileAppMetadata(this.app);
+  }
+
+  getID() {
+    return this.app.metadata.name;
+  }
+
+  setAppDetails(appName, appType, appIdentifier) {
+    this.spec.setName(appName);
+    this.spec.setType(appType);
+    this.spec.setAppIdentifier(appIdentifier);
+  }
+
+  getSpec() {
+    return new MobileAppSpec(this.app);
+  }
+
+  toJSON() {
+    return { ...this.app };
+  }
+}
+
+export class MobileAppSpec {
```

Comment:
We don't need to make those classes public. This is an internal class the the UI should not use.